### PR TITLE
dev: Add improved README

### PR DESCRIPTION
**Pull request by kfichter**
## Description
Makes the README marginally better. Might as well do it.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,74 @@
...
+`@eth-optimism/contracts` contains the various Solidity smart contracts used within the Optimistic Ethereum system.
+Some of these contracts are deployed on Ethereum ("Layer 1"), while others are meant to be deployed to Optimistic Ethereum ("Layer 2").
 
-## Building and Running
+Within each contract file you'll find a comment that lists:
+1. The compiler with which a contract is intended to be compiled, `solc` or `optimistic-solc`.
+2. The network upon to which the contract will be deployed, `OVM` or `EVM`.
 
-This package requires that `yarn` be installed on your machine!  Once it is, run `yarn test` to build and run tests.
+<!-- TODO: Add link to detailed contract spec here. -->
```
Comment (ben-chain) [inline README.md:10]: http://community.optimism.io/docs/protocol/protocol.html#system-overview for now?

Comment (kfichter) [reply README.md]: Added.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,76 @@
...
+
+```solidity
+import { SomeContract } from "@eth-optimism/contracts/SomeContract.sol";
+```
+
+## Guide for Developers
+### Setup
+Install the following:
+- [`Node.js` (14+)](https://nodejs.org/en/)
+- [`npm`](https://www.npmjs.com/get-npm)
```
Comment (tynes) [inline README.md:31]: Does this actually need both `npm` and `yarn`?

**Review by tynes — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kfichter) [reply README.md:31]: Honestly it's probably worth having both. But not sure if both are necessarily required.

**Pull request merged**